lib/src/emitter.dart

```diff
@@ -469,6 +469,13 @@ class DartEmitter extends Object
   }) {
     spec.docs.forEach(output.writeln);
     spec.annotations.forEach((a) => visitAnnotation(a, output));
+    // The `required` keyword must precede the `covariant` keyword.
```

Comment:
Do we want to guard this with nullability enabled by the emitter?

I realize it's not 1:1, just wondering.

Reply:
So, I thought of that too, but I saw that there is no prior case of the emitter throwing any exception. And other conditions, like `this.` parameters, says in the comment "Only for constructors", but this is not enforced anywhere. It seems like historically, the policy of code_builder has been "garbage in: garbage out; we're not language spec police."

Reply:
I think it's a difference case from `this.` in constructors.

In the case of `required` it's a language versioning problem. If you are otherwise able to write the exact same code between two target language versions, and only need to omit the `required` like we omit the `?` on nullable types, I think it should be controllable by the emitter.

As it stands without support for skipping this keyword in the ouput, codegen authors will need both conditionals around setting this when building the code, _and_ conditionals for enabling null safe output from the emitter.

Reply:
That makes sense. I just filed https://github.com/dart-lang/code_builder/issues/276